### PR TITLE
Fixed #24690 - Add ArrayField Mutable Default Warning

### PR DESCRIPTION
**Pull request by marissazhou**
Add ArrayField Mutable Default Warning

### CODE REVIEW

**Comment by timgraham** [comment]
The idea is that you should not use mutable defaults in the first place. For example, `default=list` instead of `default=[]`.

**Comment by mjtamlyn** [comment]
It would be useful here to explain things in a bit more detail. What happens if I actually want the default to be `['hello']` rather than an array? An answer is to pass it as `lambda: ['hello']`. This (as the `default=list` case) relies on the fact callable defaults are evaluated. An alternative would be to pass a tuple, but this means a newly instantiated object which hasn't had a round trip to the db would have an immutable value, which may easily cause bugs.

In any case, a link to the docs for field defaults seems appropriate.

File: docs/ref/contrib/postgres/fields.txt [hunk 1]
```diff
@@ -17,6 +17,27 @@ ArrayField
     <ArrayField.size>`. ``ArrayField`` can be nested to store multi-dimensional
     arrays.
 
+    .. warning::
+
+        When using a model's instance methods, it is not recommended to use
```
Comment (timgraham) [inline docs/ref/contrib/postgres/fields.txt]: The discussion of "model instance methods" seems a bit out of place as we are talking about field defaults which are specified on the model field, not on instance methods. Also, I think the Marc's suggestion about how to use non-empty defaults (with lambda's) should be included.

**Comment by timgraham** [comment]
Updated in https://github.com/django/django/pull/5078.

**Pull request closed without merging**